theano/sandbox/gpuarray/blas.py

```diff
@@ -281,7 +281,12 @@ def make_node(self, x, y):
         assert x.ndim == 2
         assert y.ndim == 2
         assert x.dtype == y.dtype
-        return Apply(self, [x, y], [x.type()])
+        otype = GpuArrayType(
+            dtype=x.dtype,
+            broadcastable=(x.type.broadcastable[0], y.type.broadcastable[1]),
+            # Here I suppose both inputs are on the same context.
+            context_name=x.type.context_name)
```

Comment:
done

On Tue, Jan 19, 2016 at 11:28 AM, abergeron <notifications@github.com>
wrote:

> In theano/sandbox/gpuarray/blas.py
> <https://github.com/Theano/Theano/pull/3881#discussion_r50135328>:
>
> > @@ -281,7 +281,12 @@ def make_node(self, x, y):
> >          assert x.ndim == 2
> >          assert y.ndim == 2
> >          assert x.dtype == y.dtype
> > -        return Apply(self, [x, y], [x.type()])
> > +        otype = GpuArrayType(
> > +            dtype=x.dtype,
> > +            broadcastable=(x.type.broadcastable[0], y.type.broadcastable[1]),
> > +            # Here I suppose both inputs are on the same context.
> > +            context_name=x.type.context_name)
>
> This would probably be better by using x.type.clone(broadcastable=...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3881/files#r50135328>.
>